ts/src/base/Exchange.ts

```diff
@@ -1043,6 +1043,10 @@ export default class Exchange {
 
     setProxyAgents (httpProxy, httpsProxy, socksProxy) {
         let chosenAgent = undefined;
+        // in browser-side, proxy modules are not supported in 'fetch/ws' methods
+        if (!isNode && (httpProxy || httpsProxy || socksProxy)) {
+            throw new NotSupported (this.id + ' - proxies in browser-side projects are not supported. You have several choices: [A] Use `exchange.proxyUrl` property to redirect through local/remote cors-proxy snippet (find sample file named "sample-local-proxy-server-with-cors" in https://github.com/ccxt/ccxt/tree/master/examples/ folder, which includes sample for REST requests) [B] override `exchange.fetch` && `exchange.watch` methods to send request through your custom setup local/remote corst proxy server');
```

Comment:
> property to redirect through local/remote cors-proxy snippet

this sentence does not make sense, what is snippet here? we should replace it with server or just remove it

Reply:
ah, it should have been 'server' not 'snippet'.  edited